src/false_positives.h

```diff
@@ -0,0 +1,118 @@
+    {0, 0, 1, 2, 4, 6, 9, 12, 15, 18, 21, 25, 28, 32, 36, 40, 44, 48, 52, 56, 61, 65, 69, 74, 79, 83, 88, 93, 97, 102, 107, 112, 117, 122, 127, 132, 138, 143, 148, 153, 159, 164, 169, 175, 180, 186, 191, 197, 202, 208, 214, 219, 225, 231, 237, 242, 248, 254, 260, 266, 272, 278, 284, 290, 296, 302, 308, 314, 320, 326, 332, 338, 344, 351, 357, 363, 369, 376, 382, 388, 394, 401, 407, 414, 420, 426, 433, 439, 446, 452, 459, 465, 472, 478, 485, 491, 498, 505, 511, 518, 524, 531, 538, 545, 551, 558, 565, 571, 578, 585, 592, 599, 605, 612, 619, 626, 633, 640, 646, 653, 660, 667, 674, 681, 688, 695, 702, 709, 716, 723, 730},
+};
+
+size_t compute_capacity(uint32_t bits, size_t max_elements, uint32_t fpbits) {
+    if (bits == 0) return 0;
+    uint16_t base_fpbits;
+    // For the purposes of table lookups, any value for bits>MAX_BITS is equal to that
+    // of bits==MAX_BITS. This is because table rows converge to floor(capacity!),
+    // and after MAX_BITS they no longer differ in ways that matter.
+    uint32_t table_bits = std::min(bits, MAX_BITS);
+    if (max_elements == 0) {
+        base_fpbits = 0;
+    } else if (bits == 1) {
+        // There is no table entry for bits==1 as it's simply a linear progression.
+        base_fpbits = max_elements - 1;
+    } else if (max_elements >= MAX_ELEM) {
+        // Conservatively evaluate at max_elements=MAX_ELEMENTS when max_elements is
+        // too large. The table is designed so that this entry is always at least
+        // 256, so up to fpbits=256, this means the return value is simply equal
+        // to max_elements. If fpbits>256, we may return an overestimate, but we
+        // don't guarantee accuracy in that case.
+        base_fpbits = BASE_FPBITS[table_bits - 2][MAX_ELEM];
+    } else {
+        base_fpbits = BASE_FPBITS[table_bits - 2][max_elements];
+    }
+    // The fpbits provided by the base max_elements==capacity case are sufficient.
+    if (base_fpbits >= fpbits) return max_elements;
+    // Otherwise, increment capacity by ceil(fpbits / bits) beyond that.
+    return max_elements + (fpbits - base_fpbits + bits - 1) / bits;
+}
+
+size_t compute_max_elements(uint32_t bits, size_t capacity, uint32_t fpbits) {
+    if (bits == 0) return 0;
+    // Start with max_elements=capacity, and decrease max_elements until the corresponding capacity is capacity.
+    size_t max_elements = capacity;
+    while (true) {
+        size_t capacity_for_max_elements = compute_capacity(bits, max_elements, fpbits);
+        CHECK_SAFE(capacity_for_max_elements >= capacity);
+        if (capacity_for_max_elements <= capacity) return max_elements;
```

Comment:
How these 2 lines make sense together? First you abort if `capacity_for_max_elements < capacity`, but then you consider `capacity_for_max_elements < capacity` as an option in the condition.

Reply:
This would be fine if it was a `==` check, I think.